feature-libs/checkout/base/components/components/delivery-mode/delivery-mode.component.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  OnDestroy,
+  OnInit,
+} from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute } from '@angular/router';
+import { CheckoutDeliveryModesFacade } from '@spartacus/checkout/base/root';
+import { DeliveryMode, FeatureConfigService } from '@spartacus/core';
+import { Observable, Subscription } from 'rxjs';
+import {
+  distinctUntilChanged,
+  filter,
+  map,
+  withLatestFrom,
+} from 'rxjs/operators';
+import { CheckoutConfigService } from '../../services/checkout-config.service';
+import { CheckoutStepService } from '../../services/checkout-step.service';
+
+@Component({
+  selector: 'cx-delivery-mode',
+  templateUrl: './delivery-mode.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DeliveryModeComponent implements OnInit, OnDestroy {
+  supportedDeliveryModes$: Observable<DeliveryMode[]>;
+  selectedDeliveryMode$: Observable<DeliveryMode>;
+  continueButtonPressed = false;
+
+  backBtnText = this.checkoutStepService.getBackBntText(this.activatedRoute);
+
+  deliveryModeSub: Subscription;
+
+  mode: FormGroup = this.fb.group({
+    deliveryModeId: ['', Validators.required],
+  });
+
+  constructor(
+    private fb: FormBuilder,
+    private checkoutConfigService: CheckoutConfigService,
+    private activatedRoute: ActivatedRoute,
+    protected checkoutStepService: CheckoutStepService,
+    protected checkoutDeliveryModesService: CheckoutDeliveryModesFacade,
```

Comment:
Let's keep the consistency of using the name somethingFacade: SomethingFacade from now on as seen in other places in this PR and in the codebase.

Even though there are some mentions of somethingService: SomethingFacade, we should clean that up in the future for somethingFacade: SomethingFacade

Reply:
I suggest we do this in the last PR (scheduled replenishment) to avoid unnecessary conflicts.